src/agentscope/agents/react_agent.py

```diff
@@ -109,7 +109,7 @@ def __init__(
 
         # Initialize a parser object to formulate the response from the model
         self.parser = RegexTaggedContentParser(
-            format_instruction="""Respond with specific tags as outlined below:
+            format_instruction="""Respond with specific tags as outlined below in json format:
```

Comment:
Why do we need to add the requirement "in json format" here?

For string type, the json format is 
```
<thought>'this is a test'</thought>
```
while the non-json format is 
```
<thought>this is a test</thought>
```

Reply:
Based on my testing, without adding 'in json format', `RegexTaggedContentParser` cannot successfully parse boolean variables or `None` (will treat them as strings) and hence cannot be recognized correctly by tools like 'execute_python_code'. E.g., if the `RegexTaggedContentParser` returns a string `None`, `if use_docker is None` will be false.

Reply:
Update: I have modified my PR to override `RegexTaggedContentParser`'s `format_instruction` within my example. I would caution against using `RegexTaggedContentParser` when `try_parse_json=True` without explicit instruction of adhering to JSON format in `format_instruction`, since LLMs will likely output contents in other formats and cause `json.loads` to fail. I will open a separate PR to address this issue.